extension/chrome/elements/compose-modules/compose-input-module.ts

```diff
@@ -78,13 +78,24 @@ export class ComposeInputModule extends ViewModule<ComposeView> {
     return this.view.sendBtnModule.popover.choices.richtext;
   };
 
+  public willInputLimitBeExceeded = (textToPaste: string, targetInputField: HTMLElement) => {
+    const limit = 50000;
+    const currentLength = targetInputField.innerText.length;
+    const isInputLimitExceeded = textToPaste.length + currentLength > limit;
+    return isInputLimitExceeded;
+  };
+
   private handlePaste = () => {
     this.squire.addEventListener('willPaste', (e: WillPasteEvent) => {
       const div = document.createElement('div');
       div.appendChild(e.fragment);
       const html = div.innerHTML;
       const sanitized = this.isRichText() ? Xss.htmlSanitizeKeepBasicTags(html, 'IMG-KEEP') : Xss.htmlSanitizeAndStripAllTags(html, '<br>', false);
       Xss.setElementContentDANGEROUSLY(div, sanitized); // xss-sanitized
+      if (this.willInputLimitBeExceeded(sanitized.trim(), this.squire.getRoot())) {
+        e.preventDefault();
+        return;
+      }
```

Comment:
`Xss.setElementContentDANGEROUSLY(div, sanitized); // xss-sanitized`
can also be moved after `if {...}`, right?

Reply:
right - you are correct. So that `Xss.set...` will execute only when needed.